### PR TITLE
internal/v4: implement icon and readme endpoints

### PR DESCRIPTION
**Pull request by rogpeppe**

### CODE REVIEW

File: internal/mongodoc/doc.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ type Entity struct {
 type FileId string
 
 const (
-	FileReadme FileId = "readme"
+	FileReadMe FileId = "readme"
```
Comment (mhilton) [inline internal/mongodoc/doc.go:86]: I'd claim readme was one word in this context. but I'm not going to quibble. I know whichever is chosen I'd get it wrong more than I get it right.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/466/

**Comment by mhilton** [comment]
:+1: LGTM

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

File: internal/mongodoc/doc.go (hunk 1 above)
Comment (rogpeppe) [inline internal/mongodoc/doc.go:86]: Yeah. We'll see...

**Pull request merged**